week_2/pet.rb

```diff
@@ -110,6 +114,16 @@ def self.cost_to_feed(pets, days)
   #
   # Note - Order is not important
   def self.group_by_animal_type(pets)
-    raise NotImplementedError # TODO
+    temp=pets.group_by { |i| i.animal_type_id }
+
+      pet_group = {}
+
+      temp.each do |id,objs|
+          names = []
+          objs.each { |obj| names<<obj.name }
```

Comment:
[`map`](https://www.rubyguides.com/2018/10/ruby-map-method/) can also be used here.